### PR TITLE
Fix SGR direct-color parsing issue

### PR DESCRIPTION
**Pull request by qwerasd205**
Fix for a little parsing issue I took note of [here](https://github.com/ghostty-org/ghostty/issues/2125#issuecomment-2466537683).

The disparity in behavior between `ghostty@main` and `xterm` can be seen with this reproduction script:
```sh
printf "\e[0m\nForeground:\n";
printf "\e[38:2:0:255:0mGreen\n";
printf "\e[38;2;0;255;0mGreen\n";
printf "\e[38:2:0:255:0:255mMagenta\n";
printf "\e[38;2;0;255;0;255mGreen\n";

printf "\e[0m\nBackground:\n";
printf "\e[48:2:0:255:0mGreen\n";
printf "\e[48;2;0;255;0mGreen\n";
printf "\e[48:2:0:255:0:255mMagenta\n";
printf "\e[48;2;0;255;0;255mGreen\n";

printf "\e[0m\nUnderline:\n";
printf "\e[58:2:0:255:0m\e[4mGreen\n";
printf "\e[58;2;0;255;0m\e[4mGreen\n";
printf "\e[58:2:0:255:0:255m\e[4mMagenta\n";
printf "\e[58;2;0;255;0;255m\e[4mGreen\n";

printf "\e[0m\n";
```

### Outputs:
|`xterm`|`ghostty@main`|this PR|
|-|-|-|
|image|image|image|

### CODE REVIEW

**Comment by qwerasd205** [comment]
Oh oops, I somehow left a test failure in, I must have accidentally hit undo at some point!

File: src/terminal/sgr.zig [hunk 1]
```diff
@@ -189,26 +189,39 @@ pub const Parser = struct {
...
-                        .g = @truncate(rgb[1]),
-                        .b = @truncate(rgb[2]),
-                    },
-                };
-            } else if (slice.len >= 3 and slice[1] == 5) {
-                self.idx += 2;
-                return Attribute{
-                    .@"256_fg" = @truncate(slice[2]),
-                };
+            38 => switch (slice[1]) {
```
Comment (gpanders) [inline src/terminal/sgr.zig:192]: Do we need to check that `slice.len >= 2` or is that guaranteed to be true at this point?

Comment (qwerasd205) [reply src/terminal/sgr.zig:192]: Ah, good catch thanks!

**Review by mitchellh — APPROVED** [review]
We should extract this to a comptime-based helper (so it can take the union tags). But this looks good and I'll merge as-is for now. Thanks!

**Pull request merged**